packages/malloy/src/lang/ast/types/malloy-element.ts

```diff
@@ -438,10 +449,14 @@ export class Document extends MalloyElement implements NameSpace {
   }
 
   getEntry(str: string): ModelEntry {
-    return this.documentModel[str];
+    return this.globalNameSpace.getEntry(str) ?? this.documentModel[str];
   }
 
   setEntry(str: string, ent: ModelEntry): void {
+    // TODO this error message is going to be in the wrong place everywhere...
+    if (this.globalNameSpace.getEntry(str) !== undefined) {
+      this.log(`Cannot redefine '${str}', which is in global namespace`);
+    }
```

Comment:
I think I 'fixed' this — this error message is a nice fallback error message, but places that actually do a redefine should check manually to see if it exists, so that it can attach the error to the right AST element.

Reply:
I think I made a comment above about this, not realizing that that was the point. I'll check back on that later.